src/GitHub.InlineReviews/Services/PullRequestSessionService.cs

```diff
@@ -260,6 +266,108 @@ public ITextDocument GetDocument(ITextBuffer buffer)
             return null;
         }
 
+        public virtual async Task<PullRequestDetailModel> ReadPullRequestDetail(HostAddress address, string owner, string name, int number)
+        {
+            if (readPullRequest == null)
+            {
+                readPullRequest = new Query()
+                    .Repository(Var(nameof(owner)), Var(nameof(name)))
```

Comment:
I'm curious why you're caching the compiled query here but not other queries? I've found an easy way to automatically handle the named arguments and caching if this is something users of GraphQL should be doing.

Reply:
Basically just because this was written from scratch and the other queries were just modified. We should probably be doing the same with the other queries, not sure if that should be in this PR or another?

Reply:
Can I show you my GraphQL prototype before we do the the same with the other queries? I'd like to find out if you think it's a good idea in principle before developing it any further. So maybe push this to another PR.

Reply:
I wanted to get your thoughts on this:
https://github.com/grokys/Octokit.GraphQL/pull/104